pkg/iac/scanners/terraform/parser/parser_test.go

```diff
@@ -905,6 +906,114 @@ data "http" "example" {
 
 func TestForEach(t *testing.T) {
 
+	t.Run("arg is set and ref to each.key", func(t *testing.T) {
+		src := `locals {
+	buckets = ["bucket1"]
+}
+
+resource "aws_s3_bucket" "this" {
+	for_each = toset(local.buckets)
+	bucket = each.key
+}`
+
+		modules := parse(t, map[string]string{
+			"main.tf": src,
+		})
+		require.Len(t, modules, 1)
+
+		buckets := modules.GetResourcesByType("aws_s3_bucket")
+		assert.Len(t, buckets, 1)
+
+		bucket := buckets[0]
+		bucketName := bucket.GetAttribute("bucket").Value().AsString()
+		assert.Equal(t, "bucket1", bucketName)
+
+		assert.Equal(t, `this["bucket1"]`, bucket.NameLabel())
+	})
+
+	t.Run("arg is set and ref to each.value", func(t *testing.T) {
+		src := `locals {
+	buckets = ["bucket1"]
+}
+
+resource "aws_s3_bucket" "this" {
+	for_each = toset(local.buckets)
+	bucket = each.value
+}`
+
+		modules := parse(t, map[string]string{
+			"main.tf": src,
+		})
+		require.Len(t, modules, 1)
+
+		buckets := modules.GetResourcesByType("aws_s3_bucket")
+		assert.Len(t, buckets, 1)
+
+		bucket := buckets[0]
+		bucketName := bucket.GetAttribute("bucket").Value().AsString()
+		assert.Equal(t, "bucket1", bucketName)
+
+		assert.Equal(t, `this["bucket1"]`, bucket.NameLabel())
+	})
+
+	t.Run("arg is map and ref to each.key", func(t *testing.T) {
+		src := `locals {
+	buckets = {
+		bucket1 = "bucket1"
```

Comment:
```suggestion
		bucket1key = "bucket1val"
```

Should we make the assertion a little easier to understand? If it's a map, the each object returns a `each.key` object returns a key and a `each.value` returns the value. In this case the assertion would strictly look for the following:

```golang
		assert.Equal(t, `this["bucket1key"]`, bucket.NameLabel())
```

as an example.

Reply:
Make sense. Done https://github.com/aquasecurity/trivy/pull/6151/commits/56e86aa0eec65e0a379bd8223d88b8bf29dae8d2